pkg/costattribution/manager.go

```diff
@@ -0,0 +1,159 @@
+package costattribution
+
+import (
+	"context"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/dskit/services"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/grafana/mimir/pkg/util/validation"
+)
+
+const (
+	missingValue  = "__missing__"
+	overflowValue = "__overflow__"
+)
+
+type Manager struct {
+	services.Service
+	logger          log.Logger
+	inactiveTimeout time.Duration
+	limits          *validation.Overrides
+	cooldownTimeout time.Duration
+
+	// mu protects the trackersByUserID map
+	tlock            sync.RWMutex
```

Comment:
Same here, let's use mutex instead of lock, and there's only one mutex here so this should be a good start:
```suggestion
	mtx            sync.RWMutex
```

Reply:
addressed here https://github.com/grafana/mimir/pull/9733/commits/f4ebc07871ef79a0ed7ee31c48032a9fd42fe757#diff-a641d740358ed8a57eec94fef0937c3e776ecc163758afc443d6b500134ccaf4L29